Assets/MRTK/SDK/Experimental/Drawstring/Scripts/TwoHandleManipulator.cs

```diff
@@ -0,0 +1,937 @@
+        public float HandleDistance => (leftHandle.localPosition - rightHandle.localPosition).magnitude;
+
+        #endregion Serialized Fields
+
+        #region Event handlers
+        [Header("Manipulation Events")]
+        [SerializeField]
+        [FormerlySerializedAs("OnManipulationStarted")]
+        private ManipulationEvent onManipulationStarted = new ManipulationEvent();
+
+        /// <summary>
+        /// Unity event raised on manipulation started
+        /// </summary>
+        public ManipulationEvent OnManipulationStarted
+        {
+            get => onManipulationStarted;
+            set => onManipulationStarted = value;
+        }
+
+        [SerializeField]
+        [FormerlySerializedAs("OnManipulationEnded")]
+        private ManipulationEvent onManipulationEnded = new ManipulationEvent();
+
+        /// <summary>
+        /// Unity event raised on manipulation ended
+        /// </summary>
+        public ManipulationEvent OnManipulationEnded
+        {
+            get => onManipulationEnded;
+            set => onManipulationEnded = value;
+        }
+
+        [SerializeField]
+        [FormerlySerializedAs("OnHoverEntered")]
+        private ManipulationEvent onHoverEntered = new ManipulationEvent();
+
+        /// <summary>
+        /// Unity event raised on hover started
+        /// </summary>
+        public ManipulationEvent OnHoverEntered
+        {
+            get => onHoverEntered;
+            set => onHoverEntered = value;
+        }
+
+        [SerializeField]
+        [FormerlySerializedAs("OnHoverExited")]
+        private ManipulationEvent onHoverExited = new ManipulationEvent();
+
+        /// <summary>
+        /// Unity event raised on hover ended
+        /// </summary>
+        public ManipulationEvent OnHoverExited
+        {
+            get => onHoverExited;
+            set => onHoverExited = value;
+        }
+        #endregion
+
+        #region Private Properties
+
+        private ManipulationMoveLogic moveLogic;
+        private TwoHandScaleLogic scaleLogic;
+        private TwoHandRotateLogic rotateLogic;
+        private ElasticHandleLogic elasticLogic;
+
+        private Vector3 leftLineOffset, rightLineOffset;
+
+        private float lastUpdateTime = 0.0f;
+
+
+        /// <summary>
+        /// Holds the pointer and the initial intersection point of the pointer ray 
+        /// with the object on pointer down in pointer space
+        /// </summary>
+        private struct PointerData
+        {
+            public IMixedRealityPointer pointer;
+            public HandleSide side;
+            private Vector3 initialGrabPointInPointer;
+
+            public PointerData(IMixedRealityPointer pointer, Vector3 worldGrabPoint, HandleSide side) : this()
+            {
+                this.pointer = pointer;
+                this.side = side;
+                this.initialGrabPointInPointer = Quaternion.Inverse(pointer.Rotation) * (worldGrabPoint - pointer.Position);
```

Comment:
Nit: Add a comment to the line around expected transform from world to local space.